stockroom/external/importer/torchvision_importers.py

```diff
@@ -41,26 +41,61 @@ def variability_status(self):
     def __len__(self):
         return len(self.dataset)
 
+    @classmethod
+    def gen_splits(cls, torchvision_dataset, root):
+        dataset_splits = []
+        for train in [True, False]:
+            dataset = torchvision_dataset(root=root, train=train, download=True)
+            dataset_splits.append(cls(dataset, train))
+        return dataset_splits
+
 
 class Cifar10(TorchvisionCommon):
     name = 'cifar10'
 
-    def __init__(self, root, train=True):
-        dataset = datasets.CIFAR10(root=root, train=train, download=True)
-        super().__init__(dataset, train)
+    @classmethod
+    def gen_splits(cls, root):
+        dataset = datasets.CIFAR10
+        dataset_splits = super().gen_splits(dataset, root)
```

Comment:
Perhaps, make it a one-liner? Also call `cls.get_splits` instead of `super()`?
```python
return cls.gen_splits(datasets.CIFAR10, root)
```

Both are not deal-breaking problems. I think it's more readable

Reply:
Yes I agree, this makes it a lot more readable

Reply:
this should be 
``` python
  return super().gen_splits(datasets.CIFAR10, root)
```
rt?
cls will refer to the current class but we want to call the method from the base class rt?
is this how it works?

Reply:
make sense